tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -56,80 +74,99 @@ public static class TestMethodAttributeExecuteIntegration
-                        case UnitTestOutcome.Inconclusive:
-                        case UnitTestOutcome.NotRunnable:
-                            if (exception is not null)
+                            var testExceptionType = testException.GetType();
+                            var testExceptionName = testExceptionType.Name;
+                            if (testExceptionName != "UnitTestAssertException" && testExceptionName != "AssertInconclusiveException")
                             {
-                                test.SetErrorInfo(exception);
-                                test.Close(TestStatus.Fail);
+                                test.SetErrorInfo(testException);
                             }
                             else
                             {
-                                test.Close(TestStatus.Skip, TimeSpan.Zero, errorMessage);
+                                test.SetErrorInfo(testExceptionType.FullName ?? "Error", testException.Message, testException.ToString());
                             }
+                        }
 
-                            break;
-                        case UnitTestOutcome.Passed:
-                            if (exception is not null)
-                            {
-                                test.SetErrorInfo(exception);
-                                test.Close(TestStatus.Fail);
-                            }
-                            else
+                        if (!string.IsNullOrEmpty(testResult.DisplayName) && test.Name != testResult.DisplayName)
+                        {
+                            test.SetName(testResult.DisplayName);
+                            MsTestIntegration.UpdateTestParameters(test, testMethodState.TestMethod, testResult.DisplayName);
+                        }
+
+                        if (exception is not null)
+                        {
+                            test.Close(TestStatus.Fail);
+                        }
+                        else
+                        {
+                            switch (testResult.Outcome)
                             {
-                                test.Close(TestStatus.Pass);
+                                case UnitTestOutcome.Error or UnitTestOutcome.Failed or UnitTestOutcome.Timeout:
+                                    test.Close(TestStatus.Fail);
+                                    break;
+                                case UnitTestOutcome.Inconclusive or UnitTestOutcome.NotRunnable:
+                                    test.Close(TestStatus.Skip, TimeSpan.Zero, testException?.Message ?? string.Empty);
+                                    break;
+                                case UnitTestOutcome.Passed:
+                                    test.Close(TestStatus.Pass);
+                                    break;
+                                default:
+                                    Common.Log.Warning("Failed to handle the test status");
```

Comment:
Maybe include the status so we know what it was if we get logs?

```suggestion
                                    Common.Log.Warning("Failed to handle the test status: {Outcome}", testResult.Outcome);
```

Reply:
done in [f4d4b7a](https://github.com/DataDog/dd-trace-dotnet/pull/5381/commits/f4d4b7acfdfab96cfbb9680c8ef1299db40978f7)